src/extension/features/budget/category-solo-mode/index.js

```diff
@@ -18,15 +18,26 @@ export class CategorySoloMode extends Feature {
   }
 
   invoke() {
-    $('.budget-toolbar').append(
-      '<span class="ember-view"><button id="all-category-expand-button" title="Expands / collapses all categories">&#8597;</button>' +
+    var dom = $('<span class="ember-view"></span>');
+    if (this.settings.enabled.indexOf('toggle-all') !== -1) {
+      dom.append(
+        '<button id="all-category-expand-button" title="Expands / collapses all categories">&#8597;' +
+          '</button>'
+      );
+      $(dom, '#all-category-expand-button').on('click', this.toggleAllCategories);
+    }
+
+    if (this.settings.enabled.indexOf('solo-mode') !== -1) {
+      dom.append(
         '<input type="checkbox" id="cat-solo-mode" class="ember-view"' +
-        getToolkitStorageKey('catSoloMode') +
-        '><label class="ember-view button" for="cat-solo-mode" title="Only have one category opened a time">Solo Mode</label></span>'
-    );
-    $('.js-budget-table-cell-collapse').on('click', this.toggleCategory);
-    $('.budget-toolbar #cat-solo-mode').on('click', this.initializeState);
-    $('.budget-toolbar #all-category-expand-button').on('click', this.toggleAllCategories);
+          getToolkitStorageKey('catSoloMode') +
+          '><label class="ember-view button" for="cat-solo-mode" title="Only have one category opened a time">Solo Mode</label>'
```

Comment:
Title should be `Only have one master category opened at a time`. 

@philipsommer I hope you don't think we're nit-picking your PR to death. That is absolutely not the intent. It's just that when I see things, I twitch until I mention them. :)

Reply:
@dbaldon Haha,thanks for the hint, you're absolutely right. fixed it.